src/openms/source/ANALYSIS/TARGETED/MetaboTargetedAssay.cpp

```diff
@@ -0,0 +1,498 @@
+          // for heuristics use the identification with the smallest mz error (ppm)
+          double min_id_mz_error = std::numeric_limits<double>::max();
+          for (unsigned int j = 0; j != min_distance_feature->getPeptideIdentifications()[0].getHits().size(); ++j)
+          {
+            double current_id_mz_error = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("mz_error_ppm");
+            // compare the absolute error absolute error
+            if (abs(current_id_mz_error) < min_id_mz_error)
+            {
+              description = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("description");
+              sumformula = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("chemical_formula");
+              adduct = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("modifications");
+
+              // change format of description [name] to name
+              description.erase(remove_if(begin(description), end(description), [](char c) { return c == '[' || c == ']'; }), end(description));
+
+              // change format of adduct information M+H;1+ -> [M+H]1+
+              String adduct_prefix = adduct.prefix(';').trim();
+              String adduct_suffix = adduct.suffix(';').trim();
+              adduct = "[" + adduct_prefix + "]" + adduct_suffix;
+
+              min_id_mz_error = abs(current_id_mz_error);
+             }
+          }
+          // use the identification with the lowest mass deviation
+          v_description.push_back(description);
+          v_sumformula.push_back(sumformula);
+          v_adduct.push_back(adduct);
+        }
+        else
+        {
+          // count UNKNOWN via transition group counter
+          v_description.push_back(String(description + "_" + transition_group_counter));
+          v_sumformula.push_back(sumformula);
+          v_adduct.push_back(adduct);
+        }
+
+      double highest_precursor_mz = 0.0;
+      float highest_precursor_int = 0.0;
+      int highest_precursor_charge = 0;
+      MSSpectrum highest_precursor_int_spectrum;
+      MSSpectrum transition_spectrum;
+      String native_id;
+
+      // find precursor/spectrum with highest intensity precursor
+      vector <size_t> index = it->second;
+
+      for (auto index_it = index.begin(); index_it != index.end(); ++index_it)
+      {
+        const MSSpectrum &spectrum = spectra[*index_it];
+
+        // check if MS2 spectrum is empty
+        if (spectrum.empty())
+        {
+          LOG_WARN << "Empty MSMS spectrum was provided. Please hava a look. Index: " << *index_it << std::endl;
+          continue;
+        }
+
+        const vector <Precursor> &precursor = spectrum.getPrecursors();
+
+        // get m/z and intensity of precursor
+        // only spectra with precursors are in the map, therefore no need to check for their presence
```

Comment:
how do you know? how can you be sure about your input? what about checking and throw an exception?

```suggestion
        // only spectra with precursors are in the map, therefore no need to check for their presence
        if (precursor.empty()) throw ...
```